flixel/addons/display/FlxPieDial.hx

```diff
@@ -0,0 +1,281 @@
+package flixel.addons.display;
+
+import flixel.FlxSprite;
+import flixel.math.FlxPoint;
+import flixel.math.FlxVector;
+import flixel.util.FlxColor;
+import flixel.util.FlxSpriteUtil;
+import haxe.EnumTools;
+import openfl.display.BitmapData;
+import openfl.display.BitmapDataChannel;
+import openfl.display.BlendMode;
+
+/**
+ * A dynamic shape that fills up the way a pie chart does. Useful for timers and other things.
+ * @author larsiusprime
+ */
+class FlxPieDial extends FlxSprite
+{
+
+	/**A value between 0.0 (empty) and 1.0 (full)**/
+	public var amount(default, set):Float = 0;
+	
+	private var pieFrames:Int = 0;
+	
+	public function new(X:Float, Y:Float, Radius:Int, Color:FlxColor=FlxColor.WHITE, Frames:Int=36, ?Shape:FlxPieDialShape, Clockwise:Bool = true, InnerRadius:Int=0) 
+	{
+		if (Shape == null) Shape = Circle;
+		super(X, Y);
+		makePieDialGraphic(Radius, Color, Frames, Shape, Clockwise, InnerRadius);
+	}
+	
+	public function set_amount(f:Float):Float
+	{
+		if (f < 0.0) f = 0.0;
+		if (f > 1.0) f = 1.0;
+		amount = f;
+		var frame:Int = Std.int(f * pieFrames);
+		animation.frameIndex = frame;
+		return amount;
+	}
+	
+	private function makePieDialGraphic(Radius:Int, Color:FlxColor, Frames:Int, Shape:FlxPieDialShape, Clockwise:Bool, InnerRadius:Int)
+	{
+		pieFrames = Frames;
+		var key:String = "pie_dial_" + Color.toHexString() + "_" + Radius + "_" + Frames + "_" + Shape;
+		var W = Radius * 2;
+		var H = Radius * 2;
+		if (false == FlxG.bitmap.checkCache(key))
+		{
+			var bmp = makePieDialGraphicSub(Radius, Color, Frames, Shape, Clockwise, InnerRadius);
+			FlxG.bitmap.add(bmp, true, key);
+		}
+		
+		loadGraphic(key, true, W, H);
+		
+		amount = 1;
+	}
+	
+	private function makePieDialGraphicSub(Radius:Int, Color:Int, Frames:Int, Shape:FlxPieDialShape, Clockwise:Bool, InnerRadius):BitmapData
```

Comment:
This function is like 120 lines long which doesn't really help readability. Surely it could be split up a bit?